File: SAM Issue 3/updated SAM files/Python/Q03bCorrect.py
import time

theDate = ""

def toCelsius (inTemp):
    celsius = 0
    celsius = (5.0 / 9.0) * (inTemp - 32.0)
    return (celsius)

def toFahrenheit (inTemp):
    fahrenheit = ((9.0 / 5.0) * inTemp) + 32.0
    return (fahrenheit)

def waitTenSeconds ():
    time.sleep (10)

def waitTime (inSeconds):
    time.sleep (inSeconds)

print (toFahrenheit (0))
print (toCelsius (212.0))   # This line does not work properly
print ("Sleeping for 10")
waitTenSeconds()
print (toCelsius (32.0))
print ("Sleeping for 5")
waitTime (5)
print (toFahrenheit (100.0))

File: SAM Issue 3/updated SAM files/Java/Q03bCorrect.java
public class Q03b
{
    String theDate = "";
    
    private static double toCelsius (double inTemp)
    {
        double celsius = 0;
        celsius = (5.0 / 9.0) * (inTemp - 32.0);
        return celsius;
    }

    private static double toFahrenheit (double inTemp)
    {
        double fahrenheit = ((9.0 / 5.0) * inTemp) + 32.0;
        return fahrenheit;
    }
    
    private static void waitTenSeconds () throws InterruptedException
    {
          Thread.sleep(10000);
    }
    
    private static void waitTime (int inMilliseconds) throws InterruptedException
    {
           Thread.sleep(inMilliseconds);
    }
    
    public static void main(String[] args) throws InterruptedException
    {
        System.out.println(toFahrenheit(0));
        System.out.println(toCelsius (212.0));  // This line does not work properly
        System.out.println("Sleeping for 10");
        waitTenSeconds();
        System.out.println (toCelsius (32.0));
        System.out.println ("Sleeping for 5");
        waitTime(5000);
        System.out.println (toFahrenheit (100.0));
    }  
}

